### PR TITLE
Fix bug where courses were wrongly opened automatically

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

The auto course opener opened a course for an HEI where that HEI had no courses open but one of its training providers did

https://ukgovernmentdfe.slack.com/archives/CQA64BETU/p1621938356057300

## Changes proposed in this pull request

- Fix a bug in `Provider#any_courses_open_in_current_cycle?` which returned true when they had no courses
- Only consider the current provider's courses, not its ratified courses, when we make the decision to auto-open
- while we're here, include the course name in the message

### CODE REVIEW

File: app/models/provider.rb [hunk 1]
```diff
@@ -56,8 +56,10 @@ def all_courses_open_in_current_cycle?(exclude_ratified_courses: false)
     open_in_find.any? && all?(&:open_on_apply?)
   end
 
-  def any_open_courses_in_current_cycle?
-    accredited_courses.or(courses).current_cycle.exposed_in_find.any?(&:open_on_apply?)
+  def any_courses_open_in_current_cycle?(exclude_ratified_courses: false)
+    courses_to_check = exclude_ratified_courses ? courses : accredited_courses.or(courses)
+
+    courses_to_check.current_cycle.exposed_in_find.any?(&:open_on_apply?)
```
Comment (cwrw) [inline app/models/provider.rb:62]: can we use the scope `open_on_apply` which covers both exposed in find and open in apply instead?courses_to_check.current_cycle.open_on_apply.any?

**Review by cwrw — APPROVED** [review]
just the one comment

File: app/models/provider.rb [hunk 2]
```diff
@@ -51,11 +51,15 @@ def acts_as_accredited_body_in_current_cycle?
   def all_courses_open_in_current_cycle?(exclude_ratified_courses: false)
     courses_to_check = exclude_ratified_courses ? courses : accredited_courses.or(courses)
 
-    courses_to_check.current_cycle.exposed_in_find.all?(&:open_on_apply?)
+    open_in_find = courses_to_check.current_cycle.exposed_in_find
+
+    open_in_find.any? && all?(&:open_on_apply?)
```
Comment (cwrw) [inline app/models/provider.rb:56]:
```suggestion
    open_in_find.any? && open_in_find.all?(&:open_on_apply?)
```

File: app/models/provider.rb (hunk 1 above)
Comment (duncanjbrown) [reply app/models/provider.rb:62]: Good call!

File: app/models/provider.rb (hunk 2 above)
Comment (duncanjbrown) [reply app/models/provider.rb:56]: oops!

**Pull request merged**